qa-tests-backend/src/main/groovy/objects/Rsyslog.groovy

```diff
@@ -0,0 +1,56 @@
+package util
+
+import groovy.util.logging.Slf4j
+import objects.Deployment
+import objects.Service
+import orchestratormanager.OrchestratorMain
+
+@Slf4j
+class Rsyslog {
+    public static final Integer SYSLOG_PORT = 514
+    private Rsyslog() { }
+    private Service syslogSvc
+    private Deployment  deployment
+
+    static Rsyslog createRsyslog(OrchestratorMain orchestrator, String namespace) {
+        def deploymentName = "rsyslog"
+        def  rsyslog = new Rsyslog()
+
+        try {
+            rsyslog.deployment =
+                    new Deployment()
+                            .setNamespace(namespace)
+                            .setName(deploymentName)
+                            .setImage("rsyslog/syslog_appliance_alpine")
+                            .addPort(SYSLOG_PORT)
+                            .addLabel("app", deploymentName)
+            orchestrator.createDeployment(rsyslog.deployment)
+
+            rsyslog.syslogSvc = new Service("rsyslog-service", orchestrator.getNameSpace())
+                                            .addLabel("app", deploymentName)
+                                            .addPort(SYSLOG_PORT , "TCP")
+                                            .setTargetPort(SYSLOG_PORT)
+                                            .setType(Service.Type.CLUSTERIP)
+            orchestrator.createService(rsyslog.syslogSvc)
+        } catch (Exception e) {
+            log.info("Something bad happened, will run cleanup before failing", e)
+            if (rsyslog.syslogSvc) {
+                orchestrator.deleteService(rsyslog.syslogSvc.name, rsyslog.syslogSvc.namespace)
+            }
+            if (rsyslog.deployment) {
+                orchestrator.deleteDeployment(rsyslog.deployment)
+            }
+            throw e
+        }
+        return rsyslog
+    }
+
+    void tearDown(OrchestratorMain orchestrator) {
+        def imagePullSecrets = deployment.getImagePullSecret()
+        for (String secret : imagePullSecrets) {
+            orchestrator.deleteSecret(secret, deployment.namespace)
+        }
+        orchestrator.deleteService(syslogSvc.name, syslogSvc.namespace)
```

Comment:
You can if case the deleteService and deleteDeployment here and use the same method from the exception catch block above